packages/cfd/src/Components/__tests__/cfd-real-account-display.spec.js

```diff
@@ -0,0 +1,627 @@
+                                        display_name: 'Driving Licence',
+                                    },
+                                    national_identity_card: {
+                                        display_name: 'National Identity Card',
+                                    },
+                                    passport: {
+                                        display_name: 'Passport',
+                                    },
+                                },
+                                is_country_supported: 1,
+                            },
+                        },
+                    },
+                    phone_idd: '62',
+                    text: 'Indonesia',
+                    tin_format: ['^\\d{15}$'],
+                    value: 'id',
+                },
+                {
+                    identity: {
+                        services: {
+                            idv: {
+                                documents_supported: {},
+                                has_visual_sample: 0,
+                                is_country_supported: 0,
+                            },
+                            onfido: {
+                                documents_supported: {},
+                                is_country_supported: 0,
+                            },
+                        },
+                    },
+                    phone_idd: '35818',
+                    text: 'Aland Islands',
+                    value: 'ax',
+                },
+            ],
+            standpoint: {
+                financial_company: 'svg',
+                gaming_company: 'svg',
+                iom: false,
+                malta: false,
+                maltainvest: false,
+                svg: true,
+            },
+            toggleAccountsDialog: jest.fn(),
+            toggleShouldShowRealAccountsList: jest.fn(),
+        };
+    });
+
+    it('should render Synthetic, Financial & Financial STP cards with enabled buttons on DMT5 when is_logged_in=true is_eu=false', () => {
+        render(<CFDRealAccountDisplay {...props} />);
+
+        const add_real_account_buttons = screen.getAllByRole('button', { name: /add real account/i });
+
+        expect(screen.getByTestId('dt_cfd_real_accounts_display')).toBeInTheDocument();
+        expect(screen.getByText('Synthetic')).toBeInTheDocument();
+        expect(screen.getByText('Financial')).toBeInTheDocument();
+        expect(screen.getByText('Financial STP')).toBeInTheDocument();
+        expect(screen.getAllByRole('table').length).toBe(3);
+        expect(screen.getAllByRole('cell', { name: /leverage/i }).length).toBe(3);
+        expect(screen.getAllByRole('cell', { name: /margin call/i }).length).toBe(3);
+        expect(screen.getAllByRole('cell', { name: /stop out level/i }).length).toBe(3);
+        expect(screen.getAllByRole('cell', { name: /number of assets/i }).length).toBe(3);
+        expect(add_real_account_buttons.length).toBe(3);
```

Comment:
Same here, I feel like we're hardcoding the DOM itself into the test cases. I feel like we need to focus on functionality and logic more instead of layout.

Reply:
I've introduced checkAccountCardsRendering() for checking the rendering of the component in each test for different cases being tested 🙂